### PR TITLE
feat: Add Custom Prompt Support

### PR DESCRIPTION
**Pull request by m3talsmith**
### 🔧 Changes

- Adds support for Custom Prompts to Management

### 📚 References

- API this is fulfilling: https://auth0.com/docs/sign-up-prompt-customizations#-signup-prompt-entry-points

### 🔬 Testing

Unit tests added into the normal flow. Make test should cover it.

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: management/management.go [hunk 1]
```diff
@@ -168,6 +171,7 @@ func New(domain string, options ...Option) (*Management, error) {
 	m.ClientGrant = (*ClientGrantManager)(&m.common)
 	m.Connection = (*ConnectionManager)(&m.common)
 	m.CustomDomain = (*CustomDomainManager)(&m.common)
+	m.CustomPrompt = (*CustomPromptManager)(&m.common)
```
Comment (sergiught) [inline management/management.go:174]: Please correct me if I am wrong but the endpoints in question seem to be under the Prompts resource within the API Docs Structure -> https://auth0.com/docs/api/management/v2/prompts/get-partials

Having these now on a separate manager might be confusing, when I'd expect to find them under the Prompt manager as it's something that should be that resource's responsibility. 

Also within the API docs I only see GET and PUT but no DELETE or POST.

**Review by sergiught — DISMISSED** [review]
It would also be great to start draft PRs within the auth0-cli and terraform provider as well with this functionality, before we merge this PR. To validate that we're indeed building the best API possible.

File: management/prompt.go [hunk 2]
```diff
@@ -49,3 +120,54 @@ func (m *PromptManager) SetCustomText(ctx context.Context, p string, l string, b
...
+	}
+	c = &CustomPrompt{Segment: segment}
+	err = m.management.Request(ctx, "GET", m.management.URI("prompts", string(segment), "partials"), c, opts...)
+	return
+}
+
+// DeleteCustomPartials deletes custom prompt partials for a given segment.
+//
+// See: https://auth0.com/docs/sign-up-prompt-customizations#use-the-api-to-edit-custom-prompts
+func (m *PromptManager) DeleteCustomPartials(ctx context.Context, c *CustomPrompt, opts ...RequestOption) error {
```
Comment (ewanharris) [inline management/prompt.go:159]: Nice one on this, I like the idea of keeping delete even though it the API doesn't strictly offer it.

File: management/prompt.go [hunk 3]
```diff
@@ -16,6 +20,73 @@ type Prompt struct {
 	WebAuthnPlatformFirstFactor *bool `json:"webauthn_platform_first_factor,omitempty"`
 }
 
+// CustomPrompt to be used for Custom Prompt Partials.
```
Comment (ewanharris) [inline management/prompt.go:23]: I don't want to bikeshed over a name, but the [API docs](https://auth0.com/docs/api/management/v2/prompts/get-partials) don't seem to use the custom prompts naming but rather `partials`, so I'm wondering if we should match that for consistency? 

The node SDK looks to be using [partials](https://github.com/auth0/node-auth0/pull/985/files#diff-42b1a344bbd2fc34f12230a57717c19eb1157069618b09126506b5ab1fd2440fR49) and [PartialsPrompt](https://github.com/auth0/node-auth0/pull/985/files#diff-bc779360d260fbc1b28befa72be7b066034440a84d6b37758ee802ef2d6dd03bR13681) (generated from the API docs)

**Review by ewanharris — COMMENTED** [review]
Just a comment around the naming to ensure we're consistent with documentation/naming of the feature

File: management/prompt.go (hunk 3 above)
Comment (m3talsmith) [reply management/prompt.go]: I like that naming convention. Changed and pushed.

**Review by ewanharris — APPROVED** [review]
Looks good! Thanks @m3talsmith!

**Comment by Arturs1123** [comment]
Wow, Great!!!!

**Pull request merged**